### PR TITLE
send early hints

### PR DESCRIPTION
**Pull request by i110**
This PR does 2 things:

1. makes headers directives with `when: early` send 103 response before dispatching to the next handler
```yaml
paths:
  /:
    headers.add:
      header:
        - "link: </index.js>; rel=preload"
        - "link: </style.css>; rel=preload"
      when: early
    file.dir: /path/to/doc_root
```

2. with mruby, adds `env['rack.early_hints']`
```yaml
paths:
  /:
    mruby.handler: |
      proc {|env|
        env['rack.early_hints'].call({'link' => "</index.js>; rel=preload\n</style.css>; rel=preload"})
        sleep 1
        [200, {}, ["hello"]]
      }
```

### CODE REVIEW

File: lib/handler/mruby.c [hunk 1]
```diff
@@ -209,6 +209,47 @@ static h2o_iovec_t convert_header_name_to_env(h2o_mem_pool_t *pool, const char *
...
+        if (mrb->exc != NULL)
+            goto Failed;
+
+        char *pos = RSTRING_PTR(value), *end = pos + RSTRING_LEN(value);
+        while (pos < end) {
+            char *found = memchr(pos, '\n', end - pos);
+            if (found == NULL)
+                found = end;
+            h2o_add_header_by_str(&generator->req->pool, &generator->req->res.headers, RSTRING_PTR(name), RSTRING_LEN(name), 1, NULL,
+                                  pos, found - pos);
```
Comment (kazuho) [inline lib/handler/mruby.c]: Am I correct in assuming that we already have code that handles `\n` for final responses? If yes, is the logic same as that for the final response? Is there a reason for not sharing the code?

Also, I believe that removing `\n` might be the correct solution here, see [RFC 7230 section 3.2.4](https://tools.ietf.org/html/rfc7230#section-3.2.4).

Comment (i110) [reply lib/handler/mruby.c]: @kazuho Thank you for the comment. I fixed it at https://github.com/h2o/h2o/pull/1767/commits/e5f3a7b761b10248e8c151e3b8133f683fdf7c16

> Also, I believe that removing \n might be the correct solution here, see RFC 7230 section 3.2.4.

In Ruby web frameworks, `\n` is widely used to indicate the frameworks to send multiple headers with a same name at once.

**Comment by kazuho** [comment]
LGTM.

**Comment by kazuho** [comment]
@i110 Do you have any open issues regarding this PR? If not, please merge, then compose a new PR that updates the documentation. Thank you in advance.

**Pull request merged**